ckanext/ontario_theme/anti_csrf3.py

```diff
@@ -0,0 +1,121 @@
+import anti_csrf
+import webob
+from webob.exc import HTTPForbidden
+
+from flask import Flask, make_response, render_template, request
+from ckan.common import config, is_flask_request
+import ckan.lib.base as base
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+CSRF_ERR = 'CSRF authentication failed. Token missing or invalid.'
+
+domain = config.get('ckan.ontario_theme.csrf_domain', '')
```

Comment:
Referrer checks should probably be optional, but if they're going to be implemented, shouldn't they be derived from `ckan.site_url`?